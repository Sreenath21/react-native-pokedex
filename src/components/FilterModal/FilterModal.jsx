import React, {useContext} from 'react';
import {View, Text, Modal, Pressable, ScrollView} from 'react-native';

import AccordionItem from '../AccordionItem/AccordionItem';
import CheckBoxContainer from '../CheckBoxContainer/CheckBoxContainer';
import {TypesContext} from '../../context/TypesContext';
import styles from './FilterModal.style';
import {
  lookForGenders,
  removeItemsFromArray,
} from '../../utils/helplerFunctions';

const FilterModal = ({modalVisible, setModalVisible}) => {
  const {
    filteredTypes,
    setFilteredTypes,
    checked,
    setChecked,
    genders,
    setGenders,
  } = useContext(TypesContext);

  // console.log('CHECKED...', checked);
  // console.log('TYPES: ', filteredTypes);
  // console.log('GENDERS: ', genders);

  const onReset = () => {
    setFilteredTypes([]);
    setChecked([]);
    setGenders([]);
    setModalVisible(false);
  };

  const onApply = () => {
    // setFilteredTypes([...checked]);
    const removedGenders = removeItemsFromArray(checked, [
      'male',
      'female',
      'genderless',
    ]);
    const onlyGenders = lookForGenders(checked);

    setFilteredTypes([...removedGenders]);
    setGenders([...onlyGenders]);
    setModalVisible(false);
  };

  const onClose = () => {
    setChecked([...filteredTypes, ...genders]);
    setModalVisible(false);
  };

  return (
    <Modal visible={modalVisible} transparent animationType="fade">
      <View style={styles.outerContainer}>
        <View style={styles.innerContainer}>
          {/* Header */}
          <View style={styles.header}>
            <Text style={styles.headerTitle}>Filters</Text>
            <Pressable style={styles.headerBtn} onPress={onClose}>
              <Text style={styles.headerBtnText}>X</Text>
            </Pressable>
          </View>

          {/*  Accordion  */}
          <ScrollView
            style={styles.scrollContainer}
            showsVerticalScrollIndicator={false}>
            <AccordionItem title="Type">
              <CheckBoxContainer title="Type" />
            </AccordionItem>

            <AccordionItem title="Gender">
              <CheckBoxContainer title="Gender" />
            </AccordionItem>
          </ScrollView>
        </View>

        {/* ABSOLUTE BUTTON CONTAINER */}
        <View style={styles.btnContainer}>
          <Pressable
            testID="reset"
            style={[styles.btn, styles.btnSecondary]}
            onPress={onReset}>
            <Text style={[styles.btnText, styles.btnSecondaryText]}>Reset</Text>
          </Pressable>

          <Pressable
            testID="apply"
            style={[styles.btn, styles.btnPrimary]}
            onPress={onApply}>
            <Text style={[styles.btnText, styles.btnPrimaryText]}>Apply</Text>
          </Pressable>
        </View>
      </View>
    </Modal>
  );
};

export default FilterModal;
