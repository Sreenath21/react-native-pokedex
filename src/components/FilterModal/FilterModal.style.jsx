import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

export default StyleSheet.create({
  outerContainer: {
    marginVertical: 28,
    marginHorizontal: 24,
    backgroundColor: '#fff',
    height: '95%',
    borderRadius: 10,
  },
  innerContainer: {
    padding: 24,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 10,
    borderBottomWidth: 1,
    borderBottomColor: '#ddd',
    marginBottom: 32,
  },
  headerTitle: {
    fontSize: 24,
    fontWeight: 'bold',
    color: colors.primary,
    flex: 1,
  },
  headerBtn: {
    borderColor: colors.primary,
    borderWidth: 1,
    borderRadius: 20,
    paddingHorizontal: 8,
    paddingVertical: 2,
  },
  headerBtnText: {
    fontSize: 16,
    fontWeight: 600,
  },
  scrollContainer: {
    marginBottom: 150,
  },
  btnContainer: {
    flexDirection: 'row',
    gap: 20,
    paddingHorizontal: 20,
    paddingVertical: 20,
    position: 'absolute',
    bottom: 0,
    borderTopWidth: 1,
    borderTopColor: '#ddd',
    backgroundColor: '#fff',
  },
  btn: {
    flex: 1,
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
  },
  btnPrimary: {
    backgroundColor: colors.primary,
  },
  btnSecondary: {
    borderWidth: 1,
    borderColor: colors.primary,
  },
  btnText: {},
  btnPrimaryText: {
    color: '#fff',
  },
  btnSecondaryText: {
    color: colors.primary,
  },
});
