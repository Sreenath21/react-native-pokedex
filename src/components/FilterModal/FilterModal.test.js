import React from 'react';
import FilterModal from './FilterModal';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import TypesContextProvider from '../../context/TypesContext';

describe('FilterModal', () => {
  it('renders correctly', () => {
    const screen = render(
      <FilterModal modalVisible={false} setModalVisible={jest.fn()} />,
      {wrapper: TypesContextProvider},
    );

    expect(screen).toMatchSnapshot();
  });
});
