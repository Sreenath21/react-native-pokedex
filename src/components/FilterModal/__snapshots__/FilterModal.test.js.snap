// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterModal renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;
